src/main/java/vn/zalopay/benchmark/GRPCSampler.java

```diff
@@ -70,37 +73,53 @@ private void initGrpcClient() {
 
     @Override
     public SampleResult sample(Entry ignored) {
-        GrpcResponse grpcResponse = new GrpcResponse();
         SampleResult sampleResult = new SampleResult();
 
-        // Console prints unknown exceptions - Intercepts unknown exceptions and sends them to the console to print instead of throwing the results into the GRPC response.
+        // Intercepts exceptions before GRPC requests are initiated
         try {
+            sampleResult.setSampleLabel(getName());
             initGrpcConfigRequest();
             initGrpcClient();
-            sampleResult.setSampleLabel(getName());
             String grpcRequest = clientCaller.buildRequestAndMetadata(getRequestJson(), getMetadata());
             sampleResult.setSamplerData(grpcRequest);
             sampleResult.setRequestHeaders(clientCaller.getMetadataString());
             sampleResult.sampleStart();
         } catch (Exception e) {
-            log.error("An unknown error occurred before the GRPC request was initiated, and the stack trace is as follows: ", e);
-            startSamplerToStopIfCatchException(sampleResult);
-            generateErrorResultInInitGRPCRequest(sampleResult, e);
```

Comment:
please keep this logic, since we need to have a detailed error when running in command-line mode when having an error with protoc invoker

Reply:
Hello, I threw this error message to JMeter observer, because we should distinguish those errors that belong to users themselves and those are the bugs of our plug-in. If printed in the console, users will think they are the bugs of our plug-in

Reply:
Another exception can happen is related to protoc method invokes, if the protoc can’t invoke to create the FileDescriptor, we can’t know exactly problem when generate it. This process is happening before we invoke grpcClientCall, so the observer in writer will not have any information

Reply:
The previous issue we were facing with exception in ProtocInvoker is below. I need to use protoc directly to get the detailed error 

[user@PMA__App-645 loadtest]$ /tmp/protocjar14303293916450843031/bin/protoc.exe /server/java-projects/loadtest/topup/temp/protofiles/base.proto /server/java-projects/loadtest/topup/temp/protofiles/base.proto -I/server/java-projects/loadtest  -I/tmp/polyglot-well-known-types4538070674483827585 -I/server/java-projects/loadtest/topup/temp/protofiles, --descriptor_set_out=/tmp/descriptor4351802933286958930.pb.bin --include_imports
/server/java-projects/loadtest/topup/temp/protofiles,: warning: directory does not exist.
base.proto: File not found.
topup/temp/protofiles/base.proto:5:1: Import "transfer_base.proto" was not found or had errors.
topup/temp/protofiles/base.proto:16:3: "grpc.base.RequestMetadata" is not defined.
topup/temp/protofiles/base.proto:63:3: "grpc.base.RequestMetadata" is not defined.

Reply:
hello
1. Add test cases to catch Protoc exceptions.
2. Added GUI test cases for catching Protoc exceptions (for style observation)

Note: The screenshot below contains Chinese, I tried to change the page to English, unfortunately failed.

![image](https://user-images.githubusercontent.com/39790401/178679260-e24adda7-e756-4dba-bec0-ce1bca434c73.png)

![image](https://user-images.githubusercontent.com/39790401/178680034-c2529001-cf7d-4ab2-8c83-82a283f737c2.png)

![image](https://user-images.githubusercontent.com/39790401/178680153-e20efb9d-8293-442b-ab91-6c61e782d5aa.png)

Reply:
thank @yl-yue